packages/core/src/store.ts

```diff
@@ -1,44 +1,33 @@
 import { callbacksMiddleware, controlledModeMiddleware, StateConfigMap } from './middlewares';
-import { createReducer, Handlers } from './reducer';
 import {
   createObservableEmitter, getChangedKeys,
   pipe, PipeFunc, SubscribeFunc, UnknownRecord,
 } from './utils';
 
-export interface StoreState<TState extends UnknownRecord> {
+export type UpdateStateAction<TState extends UnknownRecord> = (state: TState) => Partial<TState>;
+
+export interface Store<TState extends UnknownRecord> {
   getState(): TState;
   subscribe: SubscribeFunc<TState>;
-}
-
-export interface Store<
-  TState extends UnknownRecord,
-  THandlers extends Handlers<TState>,
-  > extends StoreState<TState> {
-  addUpdate(statePart: Partial<TState>): void;
+  addUpdate(updateAction: UpdateStateAction<TState>): void;
+  commitPropsUpdates(): void;
```

Comment:
Minor: Two commit methods still look a bit confusing to me, it is not clear when to call each of them. Maybe we should think about how to make usage/naming more obvious.

Reply:
Partially agree with you, but let's think about it separately